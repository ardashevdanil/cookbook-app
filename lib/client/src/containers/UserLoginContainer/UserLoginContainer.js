import { connect } from 'react-redux';

import checkUserPassword from '../../actions/checkUserPassword';
import pushUserData from '../../actions/pushUserData';
import UserLogin from '../../components/UserLogin/UserLogin';
import { userLogout } from '../../actions/userLogout';

const mapStateToProps = state => ({
  authorisationError: state.errors.authorisationError,
  signInError: state.errors.signInError,
  user: state.activeUser,
});

const mapDispatchToProps = dispatch => ({
  onLogin: payload => dispatch(checkUserPassword(payload)),
  onLogout: () => dispatch(userLogout()),
  onSignIn: payload => dispatch(pushUserData(payload)),
});

const UserLoginContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(UserLogin);

export default UserLoginContainer;
