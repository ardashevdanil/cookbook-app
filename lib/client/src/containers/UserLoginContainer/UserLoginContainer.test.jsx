import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import UserLoginContainer from './UserLoginContainer';

Enzyme.configure({ adapter: new Adapter() });

const initialState = {
  errors: {
    authorisationError: false,
    signInError: false,
  },
  activeUser: 'Guest',
};

const store = configureStore()(initialState);

describe('test UserLoginContainer component', () => {
  const component = Enzyme.mount(
    <MemoryRouter>
      <Provider store={store}>
        <UserLoginContainer />
      </Provider>
    </MemoryRouter>,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
