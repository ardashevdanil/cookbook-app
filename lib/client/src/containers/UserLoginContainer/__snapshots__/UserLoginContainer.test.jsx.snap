// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserLoginContainer component renders correctly 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Provider store={{...}}>
      <Connect(UserLogin)>
        <UserLogin authorisationError={false} signInError={false} user=\\"Guest\\" onLogin={[Function: onLogin]} onLogout={[Function: onLogout]} onSignIn={[Function: onSignIn]}>
          <div className=\\"UserLogin\\">
            <div className=\\"UserLogin__form\\">
              <div className=\\"UserLogin__heading\\">
                Login
              </div>
              Login for LEMON
              <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"login\\" type=\\"text\\" value=\\"\\" />
              <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"password\\" type=\\"password\\" value=\\"\\" />
              <div className=\\"UserLogin__error_hidden\\">
                Unable to login. Check login or password.
              </div>
              <div className=\\"UserLogin__button\\" onClick={[Function: bound handleLogin]}>
                LOGIN
              </div>
              <div className=\\"UserLogin__hint\\">
                Dont have an account? Sign in
              </div>
            </div>
            <div className=\\"UserLogin__form\\">
              <div className=\\"UserLogin__heading\\">
                Sign in
              </div>
              Sign in for LEMON
              <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"username\\" type=\\"text\\" value=\\"\\" />
              <input className=\\"UserLogin__input\\" onChange={[Function: onChange]} placeholder=\\"password\\" type=\\"password\\" value=\\"\\" />
              <n accept=\\"image/jpeg, image/png\\" acceptClassName=\\"UserLogin__dropzone_accept\\" className=\\"UserLogin__dropzone\\" maxSize={2097152} onDrop={[Function: bound onDrop]} rejectClassName=\\"UserLogin__dropzone_reject\\" preventDropOnDocument={true} disabled={false} disableClick={false} inputProps={{...}} multiple={true} minSize={0} getDataTransferItems={[Function: getDataTransferItems]}>
                <div className=\\"UserLogin__dropzone\\" style={{...}} onDrop={[Function: bound value]} onClick={[Function: bound value]} onDragStart={[Function: bound value]} onDragEnter={[Function: bound value]} onDragOver={[Function: bound value]} onDragLeave={[Function: bound value]} aria-disabled={false}>
                  Upload a profile image
                  <input accept=\\"image/jpeg, image/png\\" disabled={false} type=\\"file\\" style={{...}} multiple={true} onChange={[Function: bound value]} autoComplete=\\"off\\" />
                </div>
              </n>
              <div className=\\"UserLogin__error_hidden\\">
                Username is already taken.
              </div>
              <div className=\\"UserLogin__button\\" onClick={[Function: bound handleSignIn]}>
                SIGN UP
              </div>
              <div className=\\"UserLogin__hint\\">
                By joing LEMON you agree to our terms of service
              </div>
            </div>
          </div>
        </UserLogin>
      </Connect(UserLogin)>
    </Provider>
  </Router>
</MemoryRouter>"
`;
